content/en/users/tutorials/oidc-agent-fedcloudclient-terraform/_index.md

```diff
@@ -0,0 +1,510 @@
+> `vo.access.egi.eu` VO will be used, see
+> [Step 2: Enrolling to a Virtual Organisation](#step-2-enrolling-to-a-virtual-organisation)
+> to request access.
+
+### Deploying the Virtual Machine with terraform
+
+Instead of creating the server manually, it is possible to use
+[terraform with EGI Cloud Compute](../../compute/cloud-compute/openstack/#terraform).
+
+The
+[Terraform OpenStack provider](https://registry.terraform.io/providers/terraform-provider-openstack/openstack/latest/docs)
+provides official documentation.
+
+Terraform provides
+[installation instructions](https://www.terraform.io/downloads) for all usual
+platforms.
+
+Once terraform is installed locally, we will create a deployment as documented
+in the following sections.
+
+#### Setting up the environment
+
+The `OS_*` variables that will be used by terraform can be generated using
+`fedcloudclient`.
+
+```shell
+# Activating the virtual environment
+$ source ~/.virtualenvs/fedcloudclient/bin/activate
+# Exporting variable for VO and SITE to avoid having to repeat them
+$ export EGI_VO='vo.access.egi.eu'
+$ export EGI_SITE='IN2P3-IRES'
+eval `fedcloud site env`
+# Obtaining an OS_TOKEN for terraform
+# XXX this breaks using openstackclient: use fedcloudclient
+# or unset OS_TOKEN before using openstackclient
+$ export OS_TOKEN=$(fedcloud openstack token issue --site "$EGI_SITE" \
+    --vo "$EGI_VO" -j | jq -r '.[0].Result.id')
+```
+
+#### Describing the terraform variables
+
+The main terraform configuration file,
+[main.tf](#creating-the-main-terraform-deployment-file) is using variables that
+have to be described in a `vars.tf` file:
+
+```terraform
+# Terraform variables definition
+# Values to be provided in a *.tfvars file passed on the command line
+
+variable "internal_net_id" {
+  type        = string
+  description = "The id of the internal network"
+}
+
+variable "public_ip_pool" {
+  type        = string
+  description = "The name of the public IP address pool"
+}
+
+variable "image_id" {
+  type        = string
+  description = "VM image id"
+}
+
+variable "flavor_id" {
+  type        = string
+  description = "VM flavor id"
+}
+
+variable "security_groups" {
+  type        = list(string)
+  description = "List of security groups"
+}
+```
+
+The SITE and VO specific values for those variables will be
+[identified](#identifying-the-cloud-resources) and documented in a
+[`$EGI_SITE.tfvars` file](#documenting-the-cloud-resources-for-the-selected-site).
+
+#### Identifying the cloud resources
+
+Identify flavor, image, network and security groups for the site you want to
+use, using the information gathered with `fedcloudclient`.
+
+```shell
+# Selecting an image
+$ fedcloud select image --image-specs "Name =~ 'EGI.*22'"
+# Selecting a flavor
+$ fedcloud select flavor --flavor-specs "RAM>=2096" \
+    --flavor-specs "Disk > 10" --vcpus 2
+# Identifying available networks
+$ fedcloud openstack --site "$EGI_SITE" network list
```

Comment:
```suggestion
$ fedcloud openstack network list
```

Reply:
I initialy tried without but for me it doesn't work, that's why I added the `--site "$EGI_SITE"`:
![Screenshot 2022-10-20 at 09 55 17](https://user-images.githubusercontent.com/451417/196890028-43f7078b-2cf0-4e77-a9aa-420e94b7d045.png)

Is it working for you without the explicit `--site` parameter?

Reply:
For the record things like `fedcloud site env` work correctly and take into account the `EGI_SITE` and `EGI_VO` variables, it's only for `fedcloud openstack` that the `EGI_SITE` is apparently not taken into account:
![Screenshot 2022-10-20 at 10 12 21](https://user-images.githubusercontent.com/451417/196893716-4eb60e54-33be-4c3b-87e6-90ebb8f613d5.png)

Reply:
Discussed out of band with @sebastian-luna-valero , same behaviour for him: it doesn't  work without the `--site`.
It should in theory work like this, the documentation will be updated once the problem got identified/fixed.

Reply:
Rejecting suggestion for now.

Reply:
xref: https://github.com/tdviet/fedcloudclient/issues/150